src/analysis/convex/uniform.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import analysis.convex.strict_convex_space
+
+/-!
+# Uniformly convex spaces
+
+This file defines uniformly convex spaces, which are real normed vector spaces in which for all
+strictly positive `ε`, there exists some strictly positive `δ` such that `ε ≤ ∥x - y∥` implies
+`∥x + y∥ ≤ 2 - δ` for all `x` and `y` of norm at most than `1`. This means that the triangle
+inequality is strict with a uniform bound, as opposed to strictly convex spaces where the triangle
+inequality is strict but not necessarily uniformly (`∥x + y∥ < ∥x∥ + ∥y∥` for all `x` and `y` not in
+the same ray).
+
+## Main declarations
+
+`uniform_convex_space E` means that `E` is a uniformly convex space.
+
+## TODO
+
+* Milman-Pettis
+* Hanner's inequalities
+
+## Tags
+
+convex, uniformly convex
+-/
+
+open set metric
+open_locale convex pointwise
+
+/-- A *uniformly convex space* is a real normed space where .
+
+See also `uniform_convex_space.of_uniform_convex_closed_unit_ball`. -/
+class uniform_convex_space (E : Type*) [semi_normed_group E] : Prop :=
+(uniform_convex : ∀ ⦃ε : ℝ⦄, 0 < ε → ∃ δ, 0 < δ ∧
+  ∀ ⦃x : E⦄, ∥x∥ = 1 → ∀ ⦃y⦄, ∥y∥ = 1 → ε ≤ ∥x - y∥ → ∥x + y∥ ≤ 2 - δ)
+
+variables {E : Type*}
+
+section semi_normed_group
+variables (E) [semi_normed_group E] [uniform_convex_space E] {ε : ℝ}
+
+lemma exists_forall_sphere_dist_add_le_two_sub (hε : 0 < ε) :
+  ∃ δ, 0 < δ ∧ ∀ ⦃x : E⦄, ∥x∥ = 1 → ∀ ⦃y⦄, ∥y∥ = 1 → ε ≤ ∥x - y∥ → ∥x + y∥ ≤ 2 - δ :=
+uniform_convex_space.uniform_convex hε
+
+variables [normed_space ℝ E]
+
+lemma exists_forall_sphere_dist_add_le_two_mul_sub (hε : 0 < ε) (r : ℝ) :
+  ∃ δ, 0 < δ ∧ ∀ ⦃x : E⦄, ∥x∥ = r → ∀ ⦃y⦄, ∥y∥ = r → ε ≤ ∥x - y∥ → ∥x + y∥ ≤ 2 * r - δ :=
```

Comment:
You don't use this lemma in the next one, so you can move it after `exists_forall_ball_dist_add_le_two_mul_sub` and easily deduce it from that lemma.

Reply:
Hmmm... yeah but actually once you use the characteristic of convexity, this is really quite different.

Reply:
I will delete it for now.